src/conn/conn_api.c

```diff
@@ -1671,12 +1671,23 @@ __conn_statistics_config(WT_SESSION_IMPL *session, const char *cfg[])
 
 	if ((ret = __wt_config_subgets(
 	    session, &cval, "all", &sval)) == 0 && sval.val != 0) {
-		LF_SET(WT_CONN_STAT_ALL | WT_CONN_STAT_FAST);
+		LF_SET(WT_CONN_STAT_ALL |
+		    WT_CONN_STAT_CACHE_WALK | WT_CONN_STAT_FAST);
 		++set;
 	}
 	WT_RET_NOTFOUND_OK(ret);
 
 	if ((ret = __wt_config_subgets(
+	    session, &cval, "cache_walk", &sval)) == 0 && sval.val != 0) {
+		if (flags == 0)
+			WT_RET_MSG(session, EINVAL,
+			    "the value \"cache_walk\" is not compatible with "
```

Comment:
If I provide "cache_walk" only, and don't necessarily provide "none", it will still get set to none and error out saying "not compatible with none statistics configuration". Do you instead want to say on the lines of "cache_walk can only be provided with either fast or all".